jax/_src/pallas/mosaic/core.py

```diff
@@ -224,10 +224,9 @@ def discharges_effect(self, effect: jax_core.Effect):
 def create_tensorcore_mesh(
     axis_name: str, devices: Sequence[jax.Device] | None = None
 ) -> TensorCoreMesh:
-  # TODO(b/355036384): emit a better error if we don't have tensorcores.
   if devices is None:
     devices = jax.devices()
-  num_cores = devices[0].num_cores
+  num_cores = getattr(devices[0], 'num_cores', 1)
```

Comment:
This change could have surprising effects if there are users accidentally calling `create_tensorcore_mesh` on CPU (or with CPU as the default device) when they meant to on TPU.  (Although probably their code will still error at some later point because of the lack of a TPU.)

Some alternatives I considered:

 1. Add a `num_cores` argument to `create_tensorcore_mesh` (and forbid passing both `devices` and `num_cores`).

 2. For now, manually creating `TensorCoreMesh` objects in my tests.

What do you think?

Reply:
Sounds good -- I added a `num_cores` argument to `create_tensorcore_mesh`.